util/jsonutil/apply.go

```diff
@@ -0,0 +1,12 @@
+package jsonutil
+
+import "encoding/json"
+
+// Apply will recursively merge a src value into the dst JSON document.
+func Apply(dst []byte, src interface{}) ([]byte, error) {
```

Comment:
Since we aren't doing something like `data.Apply(rawData)`, would it be more appropriate to include this in logic `Merge()`? It appeared to me that `applyValues` is the only function here with the potential to be recursive

Reply:
`Apply` is a convenience method that can be used in-place of `json.Marshal` but with a source document. Merge takes two already-marshalled byte slices.

Reply:
I was mainly wondering if `jsonutil.Merge(src, dst)`  would make more sense over `jsonutil.Apply(src, dst)`, since `Apply` just passes itself through to `Merge`. Could we do the marshalling in Merge without needing them separated out?

Reply:
IMO `Merge` would imply combining two same-type things (e.g. two JSON documents, two structs, arrays, etc..) so that's why I chose to make a new function `Apply`. As in: `Apply this *to* the JSON document` vs `Merge these two documents`.

The only real reason for `Apply` is it seemed to make the code a little more readable by getting rid of the marshaling noise by being able to just call `jsonutil.Apply(srcDoc, newStruct)` since that's common enough case. `Merge` could potentially be made private, but didn't seem necessary as long as they both have comments/documentation on what does what.